pkg/codegen/utilities.go

```diff
@@ -162,3 +163,8 @@ func ExpandShortEnumName(name string) string {
 	}
 	return name
 }
+
+func RespectVersion() bool {
```

Comment:
1. We don't normally do env looks ups from within the engine instead doing the lookup at the outer layer and passing the result in as an option. Do you think we should do the same for codegen?
2. Why does this need a flag anyway? It looks like we're going from the nonsense of always just setting 0.0.0 to setting a proper version? (N.B. I don't know the surrounding uses cases of this code so just let me know if I'm totally missing something here)

Reply:
1. I'm not sure where to put the options. I think providers directly import `GeneratePackage`. Adding an additional option would be a breaking change (for both internal and external users of codegen). 
2. We need the flag because when we build our own providers, we don't know the version they are being built for. We include the git commit and the timestamp as part of the version. Ideally, we could set the default to true for the introduced flag.

Reply:
Codegen has typed option bags for example: 

```
// PackageInfo tracks Python-specific information associated with a package.
type PackageInfo struct {
	// PackageName is an override for the name of the generated python package.
	PackageName string `json:"packageName,omitempty"`
	// PythonRequires determines the Python versions that the generated provider supports
	PythonRequires string            `json:"pythonRequires,omitempty"`
	Requires       map[string]string `json:"requires,omitempty"`
	// Readme contains the text for the package's README.md files.
	Readme string `json:"readme,omitempty"`
	// Optional overrides for Pulumi module names
	//
	//    { "flowcontrol.apiserver.k8s.io/v1alpha1": "flowcontrol/v1alpha1" }
	//
	ModuleNameOverrides map[string]string `json:"moduleNameOverrides,omitempty"`
	// Toggle compatibility mode for a specified target.
	Compatibility string `json:"compatibility,omitempty"`
	// Deprecated: This bool is no longer needed since all providers now use input/output classes.
	UsesIOClasses bool `json:"usesIOClasses,omitempty"`
	// Determines whether to make single-return-value methods return an output object or the single value.
	LiftSingleValueMethodReturns bool `json:"liftSingleValueMethodReturns,omitempty"`
}

```

These options are typed per language so that's a bit annoying but it is does provide an avenue for bw-compat addition of options affecting codegen.

Reply:
But these bags are passed as part of the `Package` struct, and derived from the schema. I'm trying to avoid putting a `PULUMI_RESPECT_SCHEMA_VERSION` flag in the schema itself. I could create a non-serializing flag in the `Package` struct, but it seems non-optimal.